docs/program/_index.md

```diff
@@ -3,8 +3,278 @@ title: "Program a Robot"
+
+  return createRobotClient({
+    host,
+    credential,
+    authEntity: host,
+    signalingAddress,
+    iceServers
+  });
+}
+
+async function main() {
+  // Connect to client
+  let client: Client;
+  try {
+    client = await connect();
+    console.log('connected!');
+
+    let resources = await client.resourceNames();
+    console.log('Resources:');
+    console.log(resources);
+  } catch (error) {
+    console.log(error);
+    return;
+  }
+}
+
+main();
+```
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+{{< alert title="Note" color="note" >}}
+The C++ SDK is currently in alpha.
+{{< /alert >}}
+
+```cpp {class="line-numbers linkable-line-numbers"}
+# include <string>
+# include <vector>
+
+# include <boost/optional.hpp>
+
+# include <viam/api/common/v1/common.pb.h>
+# include <viam/api/robot/v1/robot.grpc.pb.h>
+
+# include <viam/sdk/robot/client.hpp>
+# include <viam/sdk/components/camera/client.hpp>
+
+using namespace viam::sdk;
+
+int main() {
+  std::string host("ADDRESS FROM THE VIAM APP");
+  DialOptions dial_opts;
+  Credentials credentials("SECRET FROM THE VIAM APP");
+  dial_opts.set_credentials(credentials);
+  boost::optional<DialOptions> opts(dial_opts);
+  Options options(0, opts);
+
+  auto robot = RobotClient::at_address(host, options);
+
+  std::cout << "Resources:\n";
+  for (const ResourceName& resource: *robot->resource_names()) {
+    std::cout << resource.namespace_() << ":" << resource.type() << ":"
+              << resource.subtype() << ":" << resource.name() << "\n";
+  }
+
+  return 0;
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+{{% /expand%}}
+
+Save this file on your development machine with the file type of your preferred SDK.
+
+Edit the boilerplate code by [adding control logic](/program/sdks/) to make a client application that connects to your robot and controls it in the way you want.
+
+Then, [execute this program](/program/run/) on any computer which:
+
+1. has [the appropriate SDK](#requirements) installed
+2. can establish a connection to your robot through the cloud, on a local or wide area network (LAN or WAN), or [locally](/program/run/#run-code-locally)
+
+### `import`: Find Required Imports
+
+The Code Sample tab contains the required imports for the SDK code at generation.
+
+If you are building out your program further or aren't using the Code Sample tab, you can find the right libraries to import to utilize SDK methods, typing, interfaces, and utilities at the start of [each resource's API documentation](/program/sdks/resource-apis/), as well as in the individual SDK documentation sites and [on GitHub](https://github.com/viamrobotics/rdk).
+
+### `connect`: Connect with Your Robot
+
+The `connect` logic of an SDK program establishes a connection for your client application to [communicate with](/internals/robot-to-robot-comms/) the robot's `viam-server` instance.
+This section of the boilerplate code contains your robot's address, location secret, and robot secret.
```

Comment:
Isn't the location and robot secret synonymous?

Reply:
So basically it was inaccurate to include in this sentence and should have been `robot part secret` but they are two separate things as far as I can tell-- see Security tab in app for the part secrets, and code sample tab for the location secret.